### PR TITLE
Fix build errors when `Kokkos_ENABLE_OpenMP=ON`

### LINKED ISSUE #154: Unable to Build NALU-CFD with Trilinos Configured for OpenMP
Hi NALU-CFD Team, been trying to get NALU-CFD running on one of our new test beds with OpenMP enabled for Trilinos. NALU-CFD will not compile because there is a forced use of Serial host-space which gives the errors below. It would be super-nice to be able to run Trilinos with OpenMP enabled and NALU-CFD in Serial if that's what is required, is that possible at all?

```
[  1%] Building CXX object CMakeFiles/nalu.dir/src/AssembleElemSolverAlgorithm.C.o
In file included from /home/sdhammo/nalu/trilinos/master-20170618/include/Kokkos_Core.hpp:53:0,
                 from /home/sdhammo/nalu/Nalu-master/include/KokkosInterface.h:12,
                 from /home/sdhammo/nalu/Nalu-master/include/SolverAlgorithm.h:13,
                 from /home/sdhammo/nalu/Nalu-master/include/AssembleElemSolverAlgorithm.h:12,
                 from /home/sdhammo/nalu/Nalu-master/src/AssembleElemSolverAlgorithm.C:10:
/home/sdhammo/nalu/trilinos/master-20170618/include/Kokkos_Serial.hpp: In instantiation of ‘typename std::enable_if<std::is_same<TagType, void>::value>::type Kokkos::Impl::ParallelFor<FunctorType, Kokkos::TeamPolicy<Properties ...>, Kokkos::Serial>::exec(Kokkos::Impl::HostThreadTeamData&) const [with TagType = void; FunctorType = sierra::nalu::AssembleElemSolverAlgorithm::execute()::<lambda(const TeamHandleType&)>; Properties = {Kokkos::Serial}; typename std::enable_if<std::is_same<TagType, void>::value>::type = void]’:
/home/sdhammo/nalu/trilinos/master-20170618/include/Kokkos_Serial.hpp:672:7:   required from ‘void Kokkos::Impl::ParallelFor<FunctorType, Kokkos::TeamPolicy<Properties ...>, Kokkos::Serial>::execute() const [with FunctorType = sierra::nalu::AssembleElemSolverAlgorithm::execute()::<lambda(const TeamHandleType&)>; Properties = {Kokkos::Serial}]’
/home/sdhammo/nalu/trilinos/master-20170618/include/Kokkos_Parallel.hpp:190:4:   required from ‘void Kokkos::parallel_for(const ExecPolicy&, const FunctorType&, const string&, typename Kokkos::Impl::enable_if<(! Kokkos::Impl::is_integral<ExecPolicy>::value)>::type*) [with ExecPolicy = Kokkos::TeamPolicy<Kokkos::Serial>; FunctorType = sierra::nalu::AssembleElemSolverAlgorithm::execute()::<lambda(const TeamHandleType&)>; std::__cxx11::string = std::__cxx11::basic_string<char>; typename Kokkos::Impl::enable_if<(! Kokkos::Impl::is_integral<ExecPolicy>::value)>::type = void]’
/home/sdhammo/nalu/Nalu-master/src/AssembleElemSolverAlgorithm.C:140:4:   required from here
/home/sdhammo/nalu/trilinos/master-20170618/include/Kokkos_Serial.hpp:640:9: error: no match for call to ‘(const sierra::nalu::AssembleElemSolverAlgorithm::execute()::<lambda(const TeamHandleType&)>) (Kokkos::Impl::ParallelFor<sierra::nalu::AssembleElemSolverAlgorithm::execute()::<lambda(const TeamHandleType&)>, Kokkos::TeamPolicy<Kokkos::Serial>, Kokkos::Serial>::Member)’
         m_functor( Member(data,ileague,m_league) );
         ^~~~~~~~~
/home/sdhammo/nalu/Nalu-master/src/AssembleElemSolverAlgorithm.C:99:79: note: candidate: sierra::nalu::AssembleElemSolverAlgorithm::execute()::<lambda(const TeamHandleType&)>
   Kokkos::parallel_for(team_exec, [&](const sierra::nalu::TeamHandleType& team)
                                                                               ^
/home/sdhammo/nalu/Nalu-master/src/AssembleElemSolverAlgorithm.C:99:79: note:   no known conversion for argument 1 from ‘Kokkos::Impl::ParallelFor<sierra::nalu::AssembleElemSolverAlgorithm::execute()::<lambda(const TeamHandleType&)>, Kokkos::TeamPolicy<Kokkos::Serial>, Kokkos::Serial>::Member {aka Kokkos::Impl::HostThreadTeamMember<Kokkos::Serial>}’ to ‘const TeamHandleType& {aka const Kokkos::Impl::HostThreadTeamMember<Kokkos::OpenMP>&}’
```

### PR DESCRIPTION
**Pull request by sayerhs**
Closes #154.

### CODE REVIEW

File: include/KokkosInterface.h [hunk 1]
```diff
@@ -15,15 +15,17 @@ namespace sierra {
 namespace nalu {
 
 using HostSpace = Kokkos::HostSpace;
-using DeviceSpace = Kokkos::DefaultExecutionSpace;
+//!!!! Important: the 'Kokkos::Serial' here makes sure we only run serial. !!!!
+// using DeviceSpace = Kokkos::DefaultExecutionSpace;
+using DeviceSpace = Kokkos::Serial;
```
Comment (aprokop) [inline include/KokkosInterface.h]: You probably want something like
```c++
#ifdef KOKKOS_HAVE_SERIAL
using DeviceSpace = Kokkos::Serial;
#else
#error "Kokkos does not have Serial space enabled"
#endif

Comment (sayerhs) [inline include/KokkosInterface.h]: Is there a situation where Kokkos::Serial isn't enabled? Can you give me an example?

Comment (aprokop) [inline include/KokkosInterface.h]: If you configure Trilinos with `-D Kokkos_ENABLE_Serial=OFF` for example.

Comment (sayerhs) [inline include/KokkosInterface.h]: OK. Thanks... I'll check that and make sure something else in Nalu doesn't break.

Comment (sayerhs) [inline include/KokkosInterface.h]: Added the `ifdef` check. Thanks.

**Pull request merged**